### DISCUSSION TITLE
"Error while opening file" "no such file or directory"

### DISCUSSION THREAD

**Discussion by mroberts1** [quarto-dev/quarto-cli#8243] [Q&A]
### Description

I am getting the following strange popup error message:

"Error while opening file" "no such file or directory"

It started when I began rendering Observable JS libraries (d3, plot) but now happens whenever I render a file in any project.

I'm running the latest version of RStudio and Quarto for MacOS 12.7.2

**Comment by mcanouil** [comment]
You need to share something here. We cannot help you just with an error message.
How can we reproduce the error? In which context the "popup" appear.
Note that since you said "popup", the issue you are facing might be an issue with RStudio rather than Quarto.
Maybe take a look at <https://github.com/rstudio/rstudio/issues>.

Could you share a small self-contained "working" (reproducible) example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

> latest version of RStudio and Quarto 
What version specifically "latest" is not really a version I'm afraid.

---

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

![A placeholder image](https://placehold.co/600x400.png)

The end.
````
`````

**Comment by mroberts1** [comment]
Below is one example of a Quarto file that produces the error, although all files that I am rendering currently produce the same error.

Screenshot of error message:

<img width="1385" alt="Screen Shot 2024-01-12 at 09 01 08" src="https://github.com/quarto-dev/quarto-cli/assets/8925762/03704cd2-a04b-43ee-871f-d43602b6a6e8">

OS: Mac 12.7.2 (Monterey)

Versions:
RStudio: 2023.12.0+369 (2023.12.0+369)
Quarto: 1.4.540

````
---
title: ''
author: ''
format: html
page-layout: full
echo: false
observable: 'https://observablehq.com/@mbostock/randomized-flood-fill'
---

```{ojs}
viewof replay = html`<button>Replay`
```

```{ojs}
canvas = {
  replay;

  const w = width + 28, h = height;
  const context = DOM.context2d(w, h, 1);
  const image = context.getImageData(0, 0, w, h);
  const indexes = [];
  let frame = 0;

  context.canvas.style.margin = "0 -14px";
  context.canvas.style.imageRendering = "pixelated";
  visit(w >> 1, h >> 1, 128, 128, 128);

  function visit(x, y, r, g, b) {
    if (x < 0 || x >= w || y < 0 || y >= h) return;
    const i = y * w + x, j = i << 2;
	if (image.data[j + 3] === 0) {
      image.data[j + 0] = r + (Math.random() < 0.5 ? -1 : 1);
      image.data[j + 1] = g + (Math.random() < 0.5 ? -1 : 1);
      image.data[j + 2] = b + (Math.random() < 0.5 ? -1 : 1);
      image.data[j + 3] = 255;
      indexes.push(i);
    }
  }
  
  while (indexes.length) {
    const i = Math.random() * indexes.length | 0;
    const j = indexes[i];
    const k = indexes.pop();
    if (i < indexes.length) indexes[i] = k;
    
    const r = image.data[(j << 2) + 0];
    const g = image.data[(j << 2) + 1];
    const b = image.data[(j << 2) + 2];
    const x = j % w, y = j / w | 0;
    visit(x - 1, y, r, g, b);
    visit(x + 1, y, r, g, b);
    visit(x, y - 1, r, g, b);
    visit(x, y + 1, r, g, b);
    
    if (!(++frame % 1000)) {
      context.putImageData(image, 0, 0);
      yield context.canvas;
    }
  }

  context.putImageData(image, 0, 0);
  yield context.canvas;
}
```

```{ojs}
width = 1000
height = 100
```


````

**Reply by cscheid** [reply]
The icon on that dialog box is from RStudio, right?

Do you see this problem when you render from a terminal? If not, then this is an issue with RStudio: https://github.com/rstudio/rstudio

**Comment by mroberts1** [comment]
Thank you, it was resolved by closing RStudio and reopening it! Just some kind of directory snafu, as it turns out.
